sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -143,43 +202,390 @@ protected AttestationToken()
+                        this.TokenSignatureBytes.ToArray(),
+                        HashAlgorithmName.SHA256);
+                    if (signatureValidated)
+                    {
+                        actualSigner = signer;
+                    }
+                    break;
+                }
+            }
+
+            if (actualSigner != null)
+            {
+                this.CertificateThumbprint = actualSigner.SigningCertificates[0].Thumbprint;
+                this.SigningCertificate = actualSigner;
+            }
+            return actualSigner != null;
+        }
+
+        /// <summary>
+        /// Validate the common properties of the attestation token - ensure that the expiration time for the token is "reasonable", if present.
+        /// </summary>
+        /// <returns>true if the common properties are valid.</returns>
+        /// <exception cref="Exception">Thrown if the attestation token is not value.</exception>
+        private bool ValidateCommonProperties(TokenValidationOptions options)
+        {
+            DateTimeOffset timeNow = DateTimeOffset.Now;
+            if (Payload.ExpirationTime.HasValue && (options?.ValidateExpirationTime ?? true))
+            {
+                if (timeNow.CompareTo(ExpirationTime.Value) > 0)
+                {
+                    if (options?.TimeValidationSlack != 0)
+                    {
+                        var delta = timeNow.Subtract(ExpirationTime.Value);
+                        if (delta.Seconds > options?.TimeValidationSlack)
+                        {
+                            throw new Exception("Attestation token expired before the slack time provided.");
+                        }
+                    }
+                    else
+                    {
+                        throw new Exception("Attestation token expired.");
+                    }
+                }
+            }
+
+            if (Payload.NotBeforeTime.HasValue && (options?.ValidateNotBeforeTime?? true))
+            {
+                if (DateTimeOffset.Now.CompareTo(NotBeforeTime.Value) < 0)
+                {
+                    if (options?.TimeValidationSlack != 0)
+                    {
+                        var delta = timeNow.Subtract(NotBeforeTime.Value);
+                        if (Math.Abs(delta.Seconds) > options?.TimeValidationSlack)
+                        {
+                            throw new Exception("Attestation token will not become valid before the slack time provided.");
+                        }
+                    }
+                    else
+                    {
+                        throw new Exception("Attestation token is not yet valid.");
+                    }
+                }
+            }
+
+            if (Payload.Issuer != null && (options?.ValidateIssuer ?? true) && options?.ExpectedIssuer != null)
```

Comment:
Shouldn't the code throw an `ArgumentException` if `options.ValidateIssuer` is true and `options.ExpectedIssuer` is null?

Reply:
Yes.